src/test/java/net/snowflake/client/jdbc/ResultSetAsyncIT.java

```diff
@@ -169,4 +173,139 @@ public void testisClosedIsLastisAfterLast() throws SQLException {
+    prepStatement.setClob(8, clob);
+    prepStatement.setBytes(9, bytes);
+    prepStatement.setBigDecimal(10, bigDecimal);
+    prepStatement.setByte(11, oneByte);
+    prepStatement.setDate(12, date);
+    prepStatement.setTime(13, time);
+    prepStatement.setTimestamp(14, ts);
+    prepStatement.execute();
+
+    ResultSet resultSet =
+        statement.unwrap(SnowflakeStatement.class).executeAsyncQuery("select * from test_get");
+    resultSet.next();
+    assertEquals(bigInt, resultSet.getInt(1));
+    assertEquals(bigInt, resultSet.getInt("COLA"));
+    assertEquals(bigLong, resultSet.getLong(2));
+    assertEquals(bigLong, resultSet.getLong("COLB"));
+    assertEquals(bigShort, resultSet.getShort(3));
+    assertEquals(bigShort, resultSet.getShort("COLC"));
+    assertEquals(str, resultSet.getString(4));
+    assertEquals(str, resultSet.getString("COLD"));
+    Reader reader = resultSet.getCharacterStream("COLD");
+    char[] sample = new char[str.length()];
+
+    assertEquals(str.length(), reader.read(sample));
+    assertEquals(str.charAt(0), sample[0]);
+    assertEquals(str, new String(sample));
+
+    assertEquals(bigDouble, resultSet.getDouble(5), 0);
+    assertEquals(bigDouble, resultSet.getDouble("COLE"), 0);
+    assertEquals(bigFloat, resultSet.getFloat(6), 0);
+    assertEquals(bigFloat, resultSet.getFloat("COLF"), 0);
+    assertTrue(resultSet.getBoolean(7));
+    assertTrue(resultSet.getBoolean("COLG"));
+    assertEquals("hello world", resultSet.getClob("COLH").toString());
+
+    // TODO: figure out why getBytes returns an offset.
+    // assertEquals(bytes, resultSet.getBytes(9));
+    // assertEquals(bytes, resultSet.getBytes("COLI"));
+
+    DecimalFormat df = new DecimalFormat("#.00");
+    assertEquals(df.format(bigDecimal), df.format(resultSet.getBigDecimal(10)));
+    assertEquals(df.format(bigDecimal), df.format(resultSet.getBigDecimal("COLJ")));
+
+    assertEquals(oneByte, resultSet.getByte(11));
+    assertEquals(oneByte, resultSet.getByte("COLK"));
+
+    SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+    assertEquals(sdf.format(date), sdf.format(resultSet.getDate(12)));
+    assertEquals(sdf.format(date), sdf.format(resultSet.getDate("COLL")));
+    assertEquals(time, resultSet.getTime(13));
+    assertEquals(time, resultSet.getTime("COLM"));
+    assertEquals(ts, resultSet.getTimestamp(14));
+    assertEquals(ts, resultSet.getTimestamp("COLN"));
+
+    // test getObject
+    assertEquals(str, resultSet.getObject(4).toString());
+    assertEquals(str, resultSet.getObject("COLD").toString());
+
+    // test getStatement method
+    assertEquals(statement, resultSet.getStatement());
+
+    prepStatement.close();
+    statement.execute("drop table if exists table_get");
+    statement.close();
+    resultSet.close();
+    connection.close();
+  }
+
+  /** At the moment, asynchronous result sets cannot be serialized. */
+  @Test
+  public void testIsSerializable() throws SQLException {
```

Comment:
At the moment, trying to serialize a result set that has been created from a query ID does not work because there is no BaseResultSet associated with it.

Reply:
can you file a JIRA so that you can revisit later?

Reply:
Yes, created [SNOW-201599](https://snowflakecomputing.atlassian.net/browse/SNOW-201599)